shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+        return self._response(
+            next_state="set_lot",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
+                "picking": self.data.picking(picking),
+            },
+            message=message,
+        )
+
+    def _response_for_set_quantity(self, picking, line, message=None):
+        return self._response(
+            next_state="set_quantity",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
+                "picking": self.data.picking(picking),
+            },
+            message=message,
+        )
+
+    def _response_for_set_destination(self, picking, line, message=None):
+        return self._response(
+            next_state="set_destination",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
+                "picking": self.data.picking(picking),
+            },
+            message=message,
+        )
+
+    def _response_for_select_dest_package(self, picking, line, message=None):
+        # NOTE: code taken from the checkout scenario.
+        # Maybe refactor it to avoid repetitions.
+        packages = picking.move_line_ids.result_package_id
+        if not packages:
+            return self._response_for_set_quantity(
+                picking,
+                line,
+                message=self.msg_store.no_valid_package_to_select(),
+            )
+        packages_data = self.data.packages(
+            packages.with_context(picking_id=picking.id).sorted(),
+            picking=picking,
+            with_packaging=True,
+        )
+        return self._response(
+            next_state="select_dest_package",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
+                "packages": packages_data,
+                "picking": self.data.picking(picking),
+            },
+            message=message,
+        )
+
+    # ENDPOINTS
+
+    def start(self):
+        return self._response_for_select_document()
+
+    def scan_document(self, barcode):
+        """Scan a picking, a product or a packaging.
+
+        Input:
+            barcode: the barcode of a product, a packaging or a picking name
+
+        transitions:
+          - select_document: Error: barcode not found
+          - select_document: Multiple picking matching the product / packaging barcode
+          - select_line: Picking scanned, one has been found
+          - set_lot: Packaging / Product has been scanned,
+                     single correspondance. Tracked product
+          - set_quantity: Packaging / Product has been scanned,
+                          single correspondance. Not tracked product
+        """
+        self._actions_for("search")
+        handlers = [
+            self._scan_document__by_picking,
+            self._scan_document__by_product,
+            self._scan_document__by_packaging,
+        ]
```

Comment:
```suggestion
        handlers = (
            self._scan_document__by_picking,
            self._scan_document__by_product,
            self._scan_document__by_packaging,
        )
```
always use tuples when you don't expect modifications